### ISSUE TITLE
Radarr unable to read ISO, WMV or AVI files

### ISSUE THREAD

**Issue by Adman1020** [Radarr/Radarr#1336]
I have a number of existing movies that are ISO, WMV or AVI file formats. Radarr is detecting the folder and searching for a movie but not able to read the file. The log lists "[movie title] has a runtime of 0, is it a valid video file?".

Radarr will then treat this as an empty folder, adding the movie to my wanted and often finding and downloading a new version of it. However as is hasn't read the original file it then doesn't replace it with the new one, meaning I am left with duplicate movies appearing in Kodi where it reads all the file types. 

Should Radarr be able to read ISO, WMV and AVI files?

This is version 0.2.0.535.

**Comment by dhaos21** [comment]
Its not implemented yet, i'm sure Galio said a few revisions ago that only the "Popular" formats are supported at the moment.

**Comment by fryfrog** [comment]
I've had problems with .wmv and .avi I'm *pretty* sure. I think .mp4 and .mkv are the only ones that work great for me. It certainly covers 90% of my files, but a lot of older dvd that hasn't been re-released on bluray or just re-ripped to x264 is .avi. :(

**Comment by galli-leo** [comment]
@dhaos21 Correct. Isos are not as simple as just adding a file extension to the include list. We have to think about them some more. Additionally, AVIs and WMVs should be supported. If you have a message "runtime of 0", that means that media info cannot read the runtime and more often than not, it is due to a corrupt file. If the file is fine however, please run media info on it yourself (https://mediaarea.net/en/MediaInfo) and post the result here so we can debug the issue.

**Comment by regulus0** [comment]
I too have this issue. The file is in the correct naming scheme and Radarr lists the folder as 0 B in size and shows it as missing, I have a few AVI's that also have the issue, but not all of them. I used the tool you mentioned (MediaInfo). The AVIs that don't detect all are missing any media data when debugged in the MediaInfo app. Those that do detect have media data.

**Comment by sstretchh** [comment]
I am having the same problem. I have an AVI file and it radarr shows the file as missing

**Comment by gangel77** [comment]
Hi Galli-leo. My understanding is you work on this. Big thanks :) its great so far (just installed yesterday)
I too am not parsing WMV files. 
Most recent i have added with a manual mediainfo:
General
Complete name                            : T:\Movies\Watchmen (2009)\Watchmen (2009).wmv
Format                                   : Windows Media
File size                                : 7.89 GiB
Duration                                 : 3h 6mn
Overall bit rate mode                    : Variable
Overall bit rate                         : 6 057 Kbps
Maximum Overall bit rate                 : 15.2 Mbps
Encoded date                             : UTC 2009-07-14 22:56:34.486

Video
ID                                       : 2
Format                                   : VC-1
Format profile                           : Advanced@L3
Codec ID                                 : WVC1
Codec ID/Hint                            : Microsoft
Description of the codec                 : Windows Media Video 9 Advanced Profile
Duration                                 : 3h 6mn
Bit rate mode                            : Variable
Bit rate                                 : 5 340 Kbps
Width                                    : 1 280 pixels
Height                                   : 534 pixels
Display aspect ratio                     : 2.40:1
Frame rate                               : 23.976 (23976/1000) fps
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Compression mode                         : Lossy
Bits/(Pixel*Frame)                       : 0.326
Stream size                              : 6.96 GiB (88%)
Language                                 : English (US)

Audio
ID                                       : 1
Format                                   : WMA
Format profile                           : Pro
Codec ID                                 : 162
Codec ID/Info                            : Windows Media Audio
Description of the codec                 : Windows Media Audio 10 Professional - 640 kbps, 48 kHz, 5.1 channel 24 bit 1-pass CBR
Duration                                 : 3h 6mn
Bit rate mode                            : Constant
Bit rate                                 : 640 Kbps
Channel(s)                               : 6 channels
Sampling rate                            : 48.0 KHz
Bit depth                                : 24 bits
Stream size                              : 854 MiB (11%)
Language                                 : English (US)

Other
Type                                     : Time code
Format                                   : WM TC
Time code of first frame                 : 00:00:00:01

**Comment by galli-leo** [comment]
@gangel77 The problem seems to be that the runtime is written as 3h 6mn instead of 3h 6min.

**Comment by gangel77** [comment]
Thanks.. so can i adjust this? do i just manually change on mediainfo? thanks for the prmpt response :)

**Comment by galli-leo** [comment]
@gangel77 No that's not something you can change, we have to

**Comment by gangel77** [comment]
k, thanks, at least u know about it now :)

**Comment by galli-leo** [comment]
@gangel77 What is the exact error Radarr is giving you?

**Comment by gangel77** [comment]
How do I check error?  

Pretty much I set a movie to wanted, it found it and downloaded it then it is still missing even after rescan.

**Comment by galli-leo** [comment]
System -> Logs should have some warning or error.

**Comment by gangel77** [comment]
K so ill rescan then check log? Ill do it in minute

**Comment by gangel77** [comment]
Ok wtf, I just did another scan to check it and now its “downloaded”

Sorry.. ill get back when I see another wmv, as I ended up deleting about 10 because it didn’t work.. then I thought I should report it but now they are gone!

**Comment by gangel77** [comment]
K, so first movie I have tried to get done by SAB.

Downloaded, found file but “Unable to parse movie info from path /downloads/Movies/Bolt.2008.BRRip.XviD.AC3-FLAWL3SS/Bolt.2008.BRRip.XviD.AC3-FLAWL3SS/Movie/Bolt-Flawl3ss.avi

“

 

Is that the “mn’ thing again u think?

Could it just me my version of media info?

 

General

Complete name                            : Z:\complete\Movies\Bolt.2008.BRRip.XviD.AC3-FLAWL3SS\Bolt.2008.BRRip.XviD.AC3-FLAWL3SS\Movie\Bolt-Flawl3ss.avi

Format                                   : AVI

Format/Info                              : Audio Video Interleave

File size                                : 1.59 GiB

Duration                                 : 1h 36mn

Overall bit rate                         : 2 365 Kbps

Writing application                      : VirtualDubMod 1.5.10.2 (build 2542/release)

Writing library                          : VirtualDubMod build 2542/release

 

Video

ID                                       : 0

Format                                   : MPEG-4 Visual

Format profile                           : Advanced Simple@L5

Format settings, BVOP                    : 2

Format settings, QPel                    : No

Format settings, GMC                     : No warppoints

Format settings, Matrix                  : Custom

Codec ID                                 : XVID

Codec ID/Hint                            : XviD

Duration                                 : 1h 36mn

Bit rate                                 : 1 908 Kbps

Width                                    : 720 pixels

Height                                   : 400 pixels

Display aspect ratio                     : 3:2

Original display aspect ratio            : 16:9

Frame rate                               : 23.976 (23976/1000) fps

Color space                              : YUV

Chroma subsampling                       : 4:2:0

Bit depth                                : 8 bits

Scan type                                : Progressive

Compression mode                         : Lossy

Bits/(Pixel*Frame)                       : 0.276

Stream size                              : 1.28 GiB (81%)

Writing library                          : XviD 1.2.1 (UTC 2008-12-04)

 

Audio

ID                                       : 1

Format                                   : AC-3

Format/Info                              : Audio Coding 3

Mode extension                           : CM (complete main)

Format settings, Endianness              : Big

Codec ID                                 : 2000

Duration                                 : 1h 36mn

Bit rate mode                            : Constant

Bit rate                                 : 448 Kbps

Channel(s)                               : 6 channels

Channel positions                        : Front: L C R, Side: L R, LFE

Sampling rate                            : 48.0 KHz

Frame rate                               : 31.250 fps (1536 spf)

Compression mode                         : Lossy

Stream size                              : 309 MiB (19%)

Alignment                                : Split accross interleaves

Interleave, duration                     : 42 ms (1.00 video frame)

Interleave, preload duration             : 500 ms

**Comment by galli-leo** [comment]
@gangel77 No, the problem here is that Radarr cannot parse the path. Can you try renaming the movie file to Bolt.2008.BRRip.XviD.AC3-FLAWL3SS?

**Comment by gangel77** [comment]
Yep that worked.

Must be im just having bad luck!

So many captures are coming out with cd1 and cd2 which I know Radarr doesn’t handle. Don’t quite know how to fix that

**Comment by Anakim** [comment]
i have also problems with some .ts mpeg streams. I have 720i and 1080i movie files. The 1080 files are fine but my 720i movies not listed.

Here my MediaInfo of one Movie

`General
ID                                       : 2 (0x2)
Complete name                            : /media/filme/Matrix Reloaded (2003)/Matrix Reloaded (2003).ts
Format                                   : MPEG-TS
File size                                : 3.43 GiB
Duration                                 : 2h 13mn
Overall bit rate mode                    : Variable
Overall bit rate                         : 3 686 Kbps

Video
ID                                       : 767 (0x2FF)
Menu ID                                  : 9 (0x9)
Format                                   : MPEG Video
Codec ID                                 : 2
Compression mode                         : Lossy
Encryption                               : Encrypted

Audio #1
ID                                       : 768 (0x300)
Menu ID                                  : 9 (0x9)
Format                                   : MPEG Audio
Codec ID                                 : 3
Compression mode                         : Lossy
Language                                 : German
Language, more info                      : Clean effects
Encryption                               : Encrypted

Audio #2
ID                                       : 769 (0x301)
Menu ID                                  : 9 (0x9)
Format                                   : MPEG Audio
Codec ID                                 : 3
Compression mode                         : Lossy
Language                                 : English
Language, more info                      : Clean effects
Encryption                               : Encrypted

Audio #3
ID                                       : 771 (0x303)
Menu ID                                  : 9 (0x9)
Format                                   : AC-3
Format/Info                              : Audio Coding 3
Codec ID                                 : 6
Channel(s)                               : 2 channels
Compression mode                         : Lossy
Language                                 : German
Language, more info                      : Clean effects
Encryption                               : Encrypted

Text
ID                                       : 32 (0x20)-100
Menu ID                                  : 9 (0x9)
Format                                   : Teletext
Language                                 : German

Menu
ID                                       : 102 (0x66)
Menu ID                                  : 9 (0x9)
Duration                                 : 2h 13mn
List                                     : 32 (0x20)-100 (Teletext, de) / 767 (0x2FF) (MPEG Video) / 768 (0x300) (MPEG Audio, German) / 769 (0x301) (MPEG Audio, English) / 771 (0x303) (AC-3, German)
Language                                 : German /  / German / English / German`

thanks
Anakim

**Comment by randydiel** [comment]
I also have the same issue. At first I thought it was an issue with the .avi files, but after lots of troubleshooting and trial and error, I found that the issue has more to do with file and dir structures than anything else. I have the same ongoing issues with all common file formats. See the image I included. Also, I was able to find that if the files are not in folders, they are very unlikely to be detected by radarr.:
![image](https://user-images.githubusercontent.com/12990838/28741592-66fb9f08-73df-11e7-9032-7845d30c5df4.png)
As you can see, simply by changing the name and adding the year, radarr was then able to parse correctly. As I figure out more about the issue trying to get all my files detected, I will send more info.
Thanks for what looks to be turning into a great product!

**Comment by drew7721** [comment]
I have the same issue with .avi and .mp4 files if the "Analyse video files" option is OFF. Analyzing the video files does extend the importing time, especially if on a remotely mounted drive. I found that running a manual "mediainfo" for an XML output does help. I would suggest running one just for the files with extensions that are not picked up for the moment.
`mediainfo filename.avi --Output=XML --LogFile="filename.xml"`
Then with the xml file with the same name as the media file it will get picked up. Creating the .nfo file might work better as well...
`mediainfo filename.avi >> filename.nfo`

**Comment by abarbaccia** [comment]
I think the same issue occurs when trying to upgrade .ts video files (DVR recordings).

**Comment by galli-leo** [comment]
@gangel77 @drew7721 @abarbaccia or anybody else with the same issue, can you guys get me a file where Radarr reports a 0 minute runtime, but mediainfo shows the correct one?

**Comment by kirazy** [comment]
Fresh installation of Radarr on Windows 10, not detecting .avi files when doing bulk import. Let me know what you need me to send you.

**Comment by Blackkatt** [comment]
Have the same issue. iso files are not consider as movies (not read or whatever) therego not marked as "downloaded =/

**Comment by imajes** [comment]
Hey- not sure if this has been seemingly fixed since, but -

```
[/movies/S/Sin City: A Dame to Kill For (2014)/Sin City: A Dame to Kill For (2014).avi] has a runtime of 0, is it a valid video file?
```

```
% mediainfo -f Sin\ City:\ A\ Dame\ to\ Kill\ For\ \(2014\).avi 
General
Count                                    : 328
Count of stream of this kind             : 1
Kind of stream                           : General
Kind of stream                           : General
Stream identifier                        : 0
Complete name                            : Sin City: A Dame to Kill For (2014).avi
File name                                : Sin City: A Dame to Kill For (2014)
File extension                           : avi
File size                                : 854857600
File size                                : 815 MiB
File size                                : 815 MiB
File size                                : 815 MiB
File size                                : 815 MiB
File size                                : 815.3 MiB
Stream size                              : 854857600
Stream size                              : 815 MiB (100%)
Stream size                              : 815 MiB
Stream size                              : 815 MiB
Stream size                              : 815 MiB
Stream size                              : 815.3 MiB
Stream size                              : 815 MiB (100%)
Proportion of this stream                : 1.00000
File last modification date              : UTC 2014-07-18 20:51:45
File last modification date (local)      : 2014-07-18 16:51:45
```

**Comment by fryfrog** [comment]
I don't think this was a `mediainfo` limitation, I think it is just a Radarr limitation. I haven't seen or tested a `.wmv` or `.avi` in a long time, but it still doesn't like `.iso`. :)

**Comment by galli-leo** [comment]
@imajes Can you send me the file at leonardo.galli@bluewin.ch ?

**Comment by neuman1812** [comment]
Hi Just found this issue with a few Movies I recently got.  This seems to be a popular format with nzb,  as its the only place Im getting AVI files.  

anyway:  here are a few logs:

from Radarr:

```
19-11-13 20:44:07.7|Info|RefreshMovieService|Updating Info for A Christmas Carol
19-11-13 20:44:08.1|Info|DiskScanService|Scanning disk for A Christmas Carol
19-11-13 20:44:08.1|Error|DetectSample|[/media/Personal/Movies/A Christmas Carol (2009)/A.Christmas.Carol.2009.BDRip.XviD-qJl.avi] has a runtime of 0, is it a valid video file?
19-11-13 20:44:08.1|Info|DiskScanService|Completed scanning disk for A Christmas Carol
19-11-13 20:44:08.1|Info|ExistingMetadataImporter|Found 0 existing metadata files
19-11-13 20:44:08.1|Info|ExistingSubtitleImporter|Found 0 existing subtitle files
19-11-13 20:44:08.1|Info|ExistingOtherExtraImporter|Found 0 existing other extra files
19-11-13 20:44:08.1|Info|ExistingExtraFileService|Found 0 extra files
19-11-13 20:44:08.2|Info|MediaCoverService|Testing: NzbDrone.Core.Messaging.Commands.CommandQueueManager, 2111

```

Mediainfo:

```
$ mediainfo info -f A.Christmas.Carol.2009.BDRip.XviD-qJl.avi 
General
Count                                    : 328
Count of stream of this kind             : 1
Kind of stream                           : General
Kind of stream                           : General
Stream identifier                        : 0
Complete name                            : A.Christmas.Carol.2009.BDRip.XviD-qJl.avi
File name                                : A.Christmas.Carol.2009.BDRip.XviD-qJl
File extension                           : avi
File size                                : 897581056
File size                                : 856 MiB
File size                                : 856 MiB
File size                                : 856 MiB
File size                                : 856 MiB
File size                                : 856.0 MiB
Stream size                              : 897581056
Stream size                              : 856 MiB (100%)
Stream size                              : 856 MiB
Stream size                              : 856 MiB
Stream size                              : 856 MiB
Stream size                              : 856.0 MiB
Stream size                              : 856 MiB (100%)
Proportion of this stream                : 1.00000
File last modification date              : UTC 2014-07-15 00:16:37
File last modification date (local)      : 2014-07-14 20:16:37

```

**Comment by fryfrog** [comment]
What version of `libmediainfo`?

**Comment by fryfrog** [comment]
Also, the output of your `mediainfo` command doesn't actually show run time. Can you put the detailed output in there? I usually just run it w/o any arguments.

**Comment by neuman1812** [comment]
Ah sorry,  I already deleted that file and got a different version.  I tried to play that file with VLC and it would not play.  So I assumed I got a bad version.  However there were a few other movies I got that did the same thing.  All AVI files.  I've already deleted them thinking they were just bad copies.  

But just in case: I have  libmediainfo0v5  version 17.12-1ubuntu0.1  

Its an ubuntu 18.04 system.
